# GTO TESS LC PREPROCESSING FOR `LSTperiod`

This notebook preprocesses the TESS LC to generate the proper datafiles for `LSTperiodogram`.

## Modules and configuration

### Modules

In [1]:
import pandas as pd

import lightkurve as lk

### Configuration

In [2]:
# FILES AND FOLDERS:
GTO_FILE = "../data/SELECTION_GTO_objects_with_PG.csv"
TARGET_FOLDER = "./GTO_TESS_LC_LSTperiod/"


### Functions

In [3]:
def tess_lc_load(filename: str):
    '''Load the TESS LC file and returns a lightcurve object with just the three relevant columns
    Note: it seems to be necessary to drop 'nan' values for GLS to work properly.'''
    lc_lk = lk.read(filename).remove_nans()
    return lc_lk

## GTO objects file load

In [4]:
gto_df = pd.read_csv(GTO_FILE, sep=',', decimal='.')
gto_df.head()

Karmn             Name Comp    GJ  RA_J2016_deg  DE_J2016_deg  \
0  J23548+385  RX J2354.8+3831    -   NaN    358.713658     38.526340   
1  J23505-095       LP 763-012    -  4367    357.634705     -9.560964   
2  J23431+365          GJ 1289    -  1289    355.781509     36.536310   
3  J23381-162        G 273-093    -  4352    354.532687    -16.236514   
4  J23245+578       BD+57 2735    -   895    351.126628     57.853057   

      RA_J2000     DE_J2000  l_J2016_deg  b_J2016_deg  ...   T0_PG_TESS  \
0  23:54:51.46  +38:31:36.2   110.941908   -23.024449  ...  1764.609498   
1  23:50:31.64  -09:33:32.7    80.777067   -67.303426  ...  1354.108815   
2  23:43:06.31  +36:32:13.1   107.922839   -24.336479  ...  1764.717539   
3  23:38:08.16  -16:14:10.2    61.845437   -69.825220  ...  1354.111098   
4  23:24:30.51  +57:51:15.5   111.552287    -3.085183  ...  1955.800582   

  e_T0_PG_TESS  offset_PG_TESS e_offset_PG_TESS  FAP_PG_TESS  valid_PG_TESS  \
0     0.010704     8262.219751         1.365667     1.000000            1.0   
1     0.001261     7767.134654         0.094298     0.064148            1.0   
2     0.003720    16158.288258         0.164698     0.002785            1.0   
3     0.000422    30353.147900         0.175123     0.031223            1.0   
4     0.001420    84823.865767         0.391298     0.799167            1.0   

   error_PG_TESS  elapsed_time_PG_TESS  \
0            NaN            344.002685   
1            NaN            473.533042   
2            NaN            352.262793   
3            NaN            485.008036   
4            NaN            476.798646   

                                      fits_file_TESS  \
0  ../data/CARMENES_GTO_TESS_PGs/J23548+385_TESS_...   
1  ../data/CARMENES_GTO_TESS_PGs/J23505-095_TESS_...   
2  ../data/CARMENES_GTO_TESS_PGs/J23431+365_TESS_...   
3  ../data/CARMENES_GTO_TESS_PGs/J23381-162_TESS_...   
4  ../data/CARMENES_GTO_TESS_PGs/J23245+578_TESS_...   

                                       fig_file_TESS  
0  ../data/CARMENES_GTO_TESS_PGs/figures/J23548+3...  
1  ../data/CARMENES_GTO_TESS_PGs/figures/J23505-0...  
2  ../data/CARMENES_GTO_TESS_PGs/figures/J23431+3...  
3  ../data/CARMENES_GTO_TESS_PGs/figures/J23381-1...  
4  ../data/CARMENES_GTO_TESS_PGs/figures/J23245+5...  

[5 rows x 248 columns]

In [5]:
len(gto_df)

239

In [6]:
print(list(gto_df.columns))

['Karmn', 'Name', 'Comp', 'GJ', 'RA_J2016_deg', 'DE_J2016_deg', 'RA_J2000', 'DE_J2000', 'l_J2016_deg', 'b_J2016_deg', 'Ref01', 'SpT', 'SpTnum', 'Ref02', 'Teff_K', 'eTeff_K', 'logg', 'elogg', '[Fe/H]', 'e[Fe/H]', 'Ref03', 'L_Lsol', 'eL_Lsol', 'Ref04', 'R_Rsol', 'eR_Rsol', 'Ref05', 'M_Msol', 'eM_Msol', 'Ref06', 'muRA_masa-1', 'emuRA_masa-1', 'muDE_masa-1', 'emuDE_masa-1', 'Ref07', 'pi_mas', 'epi_mas', 'Ref08', 'd_pc', 'ed_pc', 'Ref09', 'Vr_kms-1', 'eVr_kms-1', 'Ref10', 'ruwe', 'Ref11', 'U_kms-1', 'eU_kms-1', 'V_kms-1', 'eV_kms-1', 'W_kms-1', 'eW_kms-1', 'Ref12', 'sa_m/s/a', 'esa_m/s/a', 'Ref13', 'SKG', 'Ref14', 'SKG_lit', 'Ref14_lit', 'Pop', 'Ref15', 'vsini_flag', 'vsini_kms-1', 'evsini_kms-1', 'Ref16', 'P_d', 'eP_d', 'Ref17', 'pEWHalpha_A', 'epEWHalpha_A', 'Ref18', 'log(LHalpha/Lbol)', 'elog(LHalpha/Lbol)', 'Ref19', '1RXS', 'CRT_s-1', 'eCRT_s-1', 'HR1', 'eHR1', 'HR2', 'eHR2', 'Flux_X_E-13_ergcm-2s-1', 'eFlux_X_E-13_ergcm-2s-1', 'LX/LJ', 'eLX/LJ', 'Ref20', 'Activity', 'Ref21', 'FUV_mag',

## Create and store the relevant files for `LSTperiod`

Notice that we use the original TESS light curves to generate the files for LSTperiod, not the data stored in the periodogram calculation result file.

In [7]:
for i in range(0, len(gto_df)):
    lc_filename = "TESS-" + gto_df.loc[i, 'Karmn'] + ".dat" # Target filename for LSTperiod
    # Read the LC file:
    lc_lk = tess_lc_load(gto_df.loc[i, 'lc_file'])
    # Generate the data:
    lc_df = pd.DataFrame({'time': lc_lk['time'].value, 'value': lc_lk['flux'].value})
    # Save the file:
    lc_df.to_csv(TARGET_FOLDER + lc_filename, sep=' ', decimal='.', header=False, index=False)
    

# Summary

**OBSERVATIONS AND CONCLUSIONS:**
- We generated the 239 files of the TESS light curves in the proper format for `LSTperiod`.